src/main/java/com/aws/greengrass/clientdevices/auth/certificate/usecases/ConfigureCustomCertificateAuthority.java

```diff
@@ -25,19 +36,45 @@
  */
 public class ConfigureCustomCertificateAuthority implements UseCases.UseCase<Void, CDAConfiguration> {
     private static final Logger logger = LogManager.getLogger(ConfigureCustomCertificateAuthority.class);
-    private final CertificateManager certificateManager;
+    private final SecurityService securityService;
+    private final CertificateStore certificateStore;
 
 
     /**
      * Configure core certificate authority.
      *
-     * @param certificateManager Instance of CertificateManager
+     * @param securityService      Security Service
+     * @param certificateStore      Helper class for managing certificate
      */
     @Inject
-    public ConfigureCustomCertificateAuthority(CertificateManager certificateManager) {
-        this.certificateManager = certificateManager;
+    public ConfigureCustomCertificateAuthority(SecurityService securityService, CertificateStore certificateStore) {
+        this.securityService =  securityService;
+        this.certificateStore = certificateStore;
     }
 
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private Pair<PrivateKey, X509Certificate[]> getCertificateChain(URI privateKeyUri, URI certChainUri) {
+        // TODO: Move retry logic out of useCases
+        RetryUtils.RetryConfig retryConfig = RetryUtils.RetryConfig.builder()
```

Comment:
If this retry eventually gives up, then what happens?

200ms * 3 may not be enough time for some providers to initalize.

Reply:
+1 to this. I've commented on this several times as well. Since this is just a refactor, we can probably tackle that in a follow up. We'll want an event listener that listens to config update events and handles retries